lib/intercept.js

```diff
@@ -256,6 +277,16 @@ function overrideClientRequest() {
       )
     }
 
+    if (typeof input === 'string' && typeof options === 'object') {
```

Comment:
Am I correct this is this a Node-10-or-later feature? I wonder, is there harm in our supporting this in earlier versions of Node? I feel like that could lead to confusing disparity between nock and real ClientRequest. Perhaps we should conditionally support the new form when running in Node 10+.

Reply:
Just checked on ClientRequest in Node master, it seems they do a `typeof` check in a similar way, and then re-assigning the input to options. I think it should be backwards compatible?
https://github.com/nodejs/node/blob/master/lib/_http_client.js#L63

Reply:
I guess the question is: what happens if you're in Node 8 and you pass a string to a real ClientRequest? The behavior from ours should be the same.

Reply:
Ah, I see. I fixed it so that the Node 10 feature only work on Node 10. But I think this introduces some other problems, like testing. How do you test a Node 10 feature if you're on Node 8? Do we have a copy of Node 10 ClientRequest & Node <10 ClientRequest in nock so that tests work regardless of which Node you're testing with?

I guess easy solution would be to instead warn if someone uses Node 10 API when running on Node <10.